profiling/src/profiling/mod.rs

```diff
@@ -658,6 +664,54 @@ impl Profiler {
         }
     }
 
+    #[cfg(feature = "timeline")]
+    pub unsafe fn collect_timeline_gc_event(
+        &self,
+        duration: i64,
+        reason: &str,
+        locals: &RequestLocals,
+    ) {
+        lazy_static! {
+            static ref TIMELINE_GC_LABELS: Vec<Label> = vec![
+                Label {
+                    key: "event",
+                    value: LabelValue::Str("gc".into()),
+                },
+                Label {
+                    key: "thread id",
+                    value: LabelValue::Str("GC".into()),
+                },
+            ];
+        }
+
+        let mut labels = Profiler::message_labels();
+        labels.extend_from_slice(&TIMELINE_GC_LABELS);
+        labels.push(Label {
+            key: "gc reason",
+            value: LabelValue::Str(String::from(reason).into()),
+        });
+        let n_labels = labels.len();
+
+        match self.send_sample(Profiler::prepare_sample_message(
+            vec![ZendFrame {
+                function: "internal|gc_collect_cycles".to_string(),
```

Comment:
In .NET the garbage collector is a separate thread not having a proper (managed) call stack. Hence that's the right thing to do in .NET, but not so in PHP.

Reply:
Currently, the stack trace for this profile type is unused by the UI, right? So it would avoid unnecessary overhead.

Reply:
We have `[truncated]` as a fake frame name already, and I think native has a few other ones using square brackets as well. If we decide to use call stacks at all, I think we could push `[gc]` or `[garbage collection]` to the top of the profiler stack (not the executor vm stack). Any thoughts on that?

Reply:
I'm not sure I understand the question. Do you suggest truncating the stack to the entrypoint of the gc in general?
Unsure about that, may be helpful to locate code exhibiting heavy GC overhead.

I don't know about unused or not in the UI.

Reply:
The PR currently pushes `internal|gc_collect_cycles` to the stack, and it's the only item. It doesn't take a full stack trace. I am suggesting:

- Since the stack trace is unused, to save on time and bytes, we shouldn't send any stack at all.
- But if we do send some kind of stack now or in the future, we should put a fake frame `[gc]` or `[garbage collection]` at the top of whatever stack we send, instead of `internal|gc_collect_cycles`.

Reply:
FYI: just found out that in the current version of the backend the stack trace for gc events will be shown, but as we do not know how much GC events there are (should be really just very few). Not sending stack frames make these events missing in the resulting pprof file, IDK if the pprof spec actually allows for "no frames".
I'd suppose we do send `[gc]` and measure how many garbage collection events there are when we have this feature out. This is really low overhead (as we omit stack walking) and when we have numbers, we can decide again if want the stack trace to show up for customers.
WDYT?